packages/assistant/src/RPA/Assistant/date_picker.py

```diff
@@ -0,0 +1,87 @@
+from flet import Dropdown, UserControl, Container, Row, Icon, PopupMenuButton, PopupMenuItem, Text
+from flet.dropdown import Option
+from flet.icons import CALENDAR_MONTH, ARROW_BACK_ROUNDED, ARROW_FORWARD_ROUNDED
+from flet import MainAxisAlignment
+from datetime import datetime, date
+from calendar import monthrange
+
+
+class DatePicker(UserControl):
+
+    def _year_increment(self, e):
+        self.selected_year += 1
+        self.header_element[1].value = self.selected_year
+        self.update()
+
+    def _year_decrement(self, e):
+        self.selected_year -= 1
+        self.header_element[1].value = self.selected_year
+        self.update()
+    
+    def _select_month(self, e):
+        months = (("Jan", "Feb", "Mar", "Apr"), ("May", "Jun", "Jul", "Aug"), ("Sep", "Oct", "Nov", "Dec"))
+        selected_month = e.control._Control__previous_children[0].value
+        self.selected_month = sum(months, ()).index(selected_month) + 1
+        self.header_element[0].content = None
+        self.header_element[2].content = None
+        self.header_element[1].value = selected_month
+        self.render_day_picker()
+
+    def _select_day(self, e):
+        self.selected_day = e.control._Control__previous_children[0].value
+        self.selected_date = date(
+            year=int(self.selected_year),
+            month=int(self.selected_month),
+            day=int(self.selected_day),
+        )
+        self.pb.value = Text(self.selected_date)
+        self.update()
+
+    def render_day_picker(self):
+        element = []
+        current_row = []
+        number_of_days = monthrange(self.selected_year, self.selected_month)[1]
+        for day in range(number_of_days):
+            if day % 7 == 0 and day != 0:
+                # element.append(PopupMenuItem(content=Row(current_row, alignment=MainAxisAlignment.SPACE_BETWEEN)))
+                # self.body_element[0].content = Row(current_row, alignment=MainAxisAlignment.SPACE_BETWEEN)
+                current_row = []
+            else:
+                current_row.append(Container(content=Text(day+1), on_click=self._select_day))
+        self.body_element[0].content = Row([Container(Text("el"))], alignment=MainAxisAlignment.SPACE_BETWEEN)
+        # self.body_element = element
+        self.update()
+
+    def __init__(self):
+        # initialise default values
+        self.selected_year = datetime.now().year
+        self.selected_month = datetime.now().month
+        self.selected_day = datetime.now().day
+        self.selected_date = None
+        months = (("Jan", "Feb", "Mar", "Apr"), ("May", "Jun", "Jul", "Aug"), ("Sep", "Oct", "Nov", "Dec"))
```

Comment:
Shall we have this as a class constant since it looks like a duplicate from `_select_month` method?

Reply:
think this is not even needed and when moving it I missed deleting the old one, will do some cleanup 👍

Reply:
done